--- ./utils.c
+++ ./utils.c
@@ -105,7 +105,8 @@
 		to_fd = open(to.p_path, O_WRONLY | O_TRUNC, 0);
 	} else
 		to_fd = open(to.p_path, O_WRONLY | O_TRUNC | O_CREAT,
-		    fs->st_mode & ~(S_ISTXT | S_ISUID | S_ISGID));
+		    fs->st_mode & ~(S_ISVTX | S_ISUID | S_ISGID));
+		/* XXX: PORT: wtf? S_ISTXT vs S_ISVTX */
 
 	if (to_fd == -1) {
 		warn("%s", to.p_path);
@@ -256,7 +257,8 @@
 	int rval;
 
 	rval = 0;
-	fs->st_mode &= S_ISTXT | S_ISUID | S_ISGID | S_IRWXU | S_IRWXG | S_IRWXO;
+	fs->st_mode &= S_ISVTX | S_ISUID | S_ISGID | S_IRWXU | S_IRWXG | S_IRWXO;
+	/* XXX: PORT: wtf? S_ISTXT vs S_ISVTX */
 
 	ts[0] = fs->st_atim;
 	ts[1] = fs->st_mtim;
@@ -277,7 +279,8 @@
 			warn("chown: %s", to.p_path);
 			rval = 1;
 		}
-		fs->st_mode &= ~(S_ISTXT | S_ISUID | S_ISGID);
+		fs->st_mode &= ~(S_ISVTX | S_ISUID | S_ISGID);
+		/* XXX: PORT: wtf? S_ISTXT vs S_ISVTX */
 	}
 	if (fd >= 0 ? fchmod(fd, fs->st_mode) :
 	    fchmodat(AT_FDCWD, to.p_path, fs->st_mode, AT_SYMLINK_NOFOLLOW)) {
@@ -285,6 +288,7 @@
 		rval = 1;
 	}
 
+#if 0
 	/*
 	 * XXX
 	 * NFS doesn't support chflags; ignore errors unless there's reason
@@ -299,6 +303,7 @@
 			warn("chflags: %s", to.p_path);
 			rval = 1;
 		}
+#endif
 	return (rval);
 }
