--- md5.c
+++ md5.c
@@ -21,8 +21,9 @@
  * Materiel Command, USAF, under agreement number F39502-99-1-0512.
  */
 
+#include <sys/time.h>
 #include <sys/types.h>
-#include <sys/queue.h>
+#include <openbsd-queue.h>
 #include <netinet/in.h>
 #include <ctype.h>
 #include <err.h>
@@ -32,7 +33,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <limits.h>
-#include <time.h>
 #include <unistd.h>
 #include <errno.h>
 
@@ -51,6 +51,8 @@
 #define MINIMUM(a, b)	(((a) < (b)) ? (a) : (b))
 #define MAXIMUM(a, b)	(((a) > (b)) ? (a) : (b))
 
+extern int optreset;
+
 union ANY_CTX {
 #if !defined(SHA2_ONLY)
 	CKSUM_CTX cksum;
