--- dd.c.orig
+++ dd.c
@@ -34,6 +34,7 @@
  * SUCH DAMAGE.
  */
 
+#include <sys/time.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
@@ -74,7 +75,7 @@
 	jcl(argv);
 	setup();
 
-	(void)signal(SIGINFO, summaryx);
+	(void)signal(SIGPWR, summaryx);
 	(void)signal(SIGINT, terminate);
 
 	atexit(summary);
--- args.c.orig
+++ args.c
@@ -167,8 +167,8 @@
 	if (cbsz > SSIZE_MAX || in.dbsz > SSIZE_MAX || out.dbsz > SSIZE_MAX)
 		errx(1, "buffer sizes cannot be greater than %zd",
 		    (ssize_t)SSIZE_MAX);
-	if (in.offset > QUAD_MAX / in.dbsz || out.offset > QUAD_MAX / out.dbsz)
-		errx(1, "seek offsets cannot be larger than %qd", QUAD_MAX);
+	if (in.offset > LLONG_MAX / in.dbsz || out.offset > LLONG_MAX / out.dbsz)
+		errx(1, "seek offsets cannot be larger than %qd", LLONG_MAX);
 }
 
 static int
@@ -407,7 +407,7 @@
 	char *expr;
 
 	num = strtoq(val, &expr, 0);
-	if (num == QUAD_MAX)			/* Overflow. */
+	if (num == LLONG_MAX)			/* Overflow. */
 		err(1, "%s", oper);
 	if (expr == val)			/* No digits. */
 		errx(1, "%s: illegal numeric value", oper);
